sphinx_immaterial/inline_icons.py

```diff
@@ -0,0 +1,57 @@
+"""This module adds the md-icon role. This is specific to this this theme because,
+primarily, the icons bundled with this theme are supported (plus any svg files in the
+user project's static directories)."""
+from pathlib import Path
+from typing import Tuple, List
+from docutils import nodes
+from sphinx.application import Sphinx
+from sphinx.writers.html5 import HTML5Translator
+
+
+class md_icon(nodes.container):
+    pass
+
+
+def visit_md_icon(self: HTML5Translator, node: md_icon):
+    """read icon data and put it into a span element"""
+    # in case this node was created from another extension
+    assert node.rawsource, "icon node's rawsource must be a relative path"
+    icon_name = node.rawsource + ".svg"
+    svg = Path(__file__).parent / ".icons" / icon_name
+    if not svg.exists():
+        static_paths: str = getattr(self.builder.config, "html_static_path")
+        if not static_paths:
+            raise FileNotFoundError(f"{icon_name} not found in bundled theme icons")
+        for path in static_paths:
+            svg = Path(self.builder.srcdir) / path / icon_name
+            if svg.exists():
+                break
+        else:
+            raise FileNotFoundError(
+                f"{icon_name} not found in html_static_path and not bundled with theme"
+            )
+    svg_data = svg.read_text(encoding="utf-8")
+
+    css_classes = " ".join(node["classes"])
+    self.body.append(f'<span class="{css_classes}">{svg_data}</span>')
```

Comment:
I was thinking about extracting this alogorithm into a separate importable function, so it could be used in other parts of the theme (eg for nav icons or #183 )

Reply:
That would make sense.  One issue I noticed is that this will duplicate the svg for each use of the icon.  If the same icon is referenced many times (e.g. as may be the case of nav icons) that will bloat the generated output.   Instead we could emit it as CSS into the css bundle, once we have that implemented, and then it would only occur once.  Or maybe could switch between the two behaviors, depending on whether the icon is used more than once.